crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -285,6 +285,9 @@ def finalize_cluster_upgrade(cluster)
         save_upgrade_state("Finalizing the upgrade of nodes in cluster #{cluster}")
```

Comment:
Are you sure this commit message is right?  The routers should be distributed fairly equally across all upgraded nodes by the end of the upgrade, because when each node is upgraded, its routers are migrated to some random other node.

Having said that, a rebalance is still a good idea, but the commit message should be fixed.